### PR TITLE
Disable MBEDTLS_SSL_RENEGOTIATION in TLS-1.3-only builds

### LINKED ISSUE #6200: Unused variable in TLS 1.3-only builds
As of [`mbedtls-3.2.1`](https://github.com/Mbed-TLS/mbedtls/tree/mbedtls-3.2.1), consider the following configuration (perfectly reasonable for a 1.3-only build):
```
#define MBEDTLS_SSL_PROTO_TLS1_3
#define MBEDTLS_SSL_TLS1_3_COMPATIBILITY_MODE

#undef MBEDTLS_SSL_DTLS_CONNECTION_ID
#undef MBEDTLS_SSL_ENCRYPT_THEN_MAC
#undef MBEDTLS_SSL_EXTENDED_MASTER_SECRET
#undef MBEDTLS_SSL_RENEGOTIATION
#undef MBEDTLS_SSL_PROTO_TLS1_2
#undef MBEDTLS_SSL_PROTO_DTLS
#undef MBEDTLS_SSL_DTLS_ANTI_REPLAY
#undef MBEDTLS_SSL_DTLS_HELLO_VERIFY
#undef MBEDTLS_SSL_DTLS_SRTP
#undef MBEDTLS_SSL_DTLS_CLIENT_PORT_REUSE
```
This is similar to a configuration you'd typically reach by enabling only TLS features that apply to TLS 1.3. The configuration from #6118 has the same problem.

Building the library produces an unused variable warning:
```
source/library/ssl_msg.c: In function ‘ssl_handle_hs_message_post_handshake’:
source/library/ssl_msg.c:5297:9: error: unused variable ‘ret’ [-Werror=unused-variable]
 5297 |     int ret = MBEDTLS_ERR_ERROR_CORRUPTION_DETECTED;
      |         ^~~
cc1: all warnings being treated as errors
```

This should have been caught by a 1.3-only on the CI (`component_test_tls13_only`), but it wasn't because `tests/configs/tls13-only.h` keeps `MBEDTLS_SSL_RENEGOTIATION` enabled even though it's a 1.2-only feature.

On a related note, if the library is built with support for both TLS 1.2 and TLS 1.3, `ssl_handle_hs_message_post_handshake()` may send an alert when the protocol is 1.3, which I don't think is intended.

### PR DESCRIPTION
**Pull request by gilles-peskine-arm**
Fixes #6200

Not applicable to 2.28.

- [x] **changelog** provided
- [x] **backport** not required
- [x] **tests** not required

### CODE REVIEW

File: tests/configs/tls13-only.h [hunk 1]
```diff
@@ -25,10 +25,13 @@
 #define MBEDTLS_SSL_PROTO_TLS1_3
 #define MBEDTLS_SSL_TLS1_3_COMPATIBILITY_MODE
 
+#undef MBEDTLS_SSL_DTLS_CONNECTION_ID
```
Comment (tom-cosgrove-arm) [inline tests/configs/tls13-only.h:28]: Are these configs ever built, e.g. by copying them to `include/mbedtls/mbedtls_config.h` and building, or are they just untested examples?

**Comment by valord577** [comment]
I also get err:

```text
[ 90%] Building C object library/CMakeFiles/mbedtls.dir/ssl_msg.c.o
/root/code/mbedtls/library/ssl_msg.c:5303:9: error: unused variable 'ret' [-Werror,-Wunused-variable]
    int ret = MBEDTLS_ERR_ERROR_CORRUPTION_DETECTED;
        ^
1 error generated.
make[2]: *** [library/CMakeFiles/mbedtls.dir/build.make:160: library/CMakeFiles/mbedtls.dir/ssl_msg.c.o] Error 1
make[2]: *** Waiting for unfinished jobs....
make[1]: *** [CMakeFiles/Makefile2:219: library/CMakeFiles/mbedtls.dir/all] Error 2
make: *** [Makefile:136: all] Error 2
```

**Review by daverodgman — APPROVED** [review]

**Comment by tom-cosgrove-arm** [comment]
@gilles-peskine-arm This PR has conflicts

**Review by mprse — COMMENTED** [review]
This PR needs to be rebased to resolve conflicts.
Please note that `component_test_psa_crypto_config_accel_hash_use_psa` needs to be keep in sync with `component_test_psa_crypto_config_reference_hash_use_psa`.

**Review by mprse — APPROVED** [review]

**Review by daverodgman — APPROVED** [review]

**Pull request merged**